File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
# DFS 와 BFS
def dfs(p):    # 재귀
    visited[p] = 1
    print(p, end=' ')
    for i in graph[p]:
        if visited[i] == 0:
            dfs(i)


def bfs(p):
    print(p, end=' ')
    visited[p] = 1
    need_visit.append(p)
    while need_visit:
        tmp = need_visit.popleft() # 맨 앞에꺼를 빼서
        for i in graph[tmp]:    # 그 인접 노드를 모두 순회
            if visited[i] == 0:
                visited[i] = 1
                print(i, end=' ')
                need_visit.append(i)


import sys
from collections import deque

n, m, v= map(int, sys.stdin.readline().strip().split()) # 정점, 간선, 탐색시작점

graph = [[] for _ in range(n+1)]

for i in range(m):
    s, e = map(int, sys.stdin.readline().strip().split())
    graph[s].append(e)
    graph[e].append(s)


for i in range(n+1):
    graph[i] = sorted(graph[i])

# 방문 체크 리스트
visited = [0] * (n+1)

need_visit = deque()

dfs(v)
print()
visited = [0] * (n+1)

bfs(v)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
function solution(n, v, graph) {
  // console.log(graph);
  let dfsAnswer = [v];
  let bfsAnswer = [];

  let dfsVisited = new Array(n + 1).fill(false);
  let bfsVisited = new Array(n + 1).fill(false);

  function dfs(n, v, graph) {
    dfsVisited[v] = true;
    // console.log(dfsAnswer);

    if (dfsAnswer.length === n) {
      return;
    } else {
      for (let j = 0; j < graph[v].length; j++) {
        if (!dfsVisited[graph[v][j]]) {
          // console.log(graph[v]);
          dfsVisited[graph[v][j]] = true;
          dfsAnswer.push(graph[v][j]);
          dfs(n, graph[v][j], graph);
        }
      }
    }
  }

  function bfs(n, v, graph) {
    let queue = [v];

    while (queue.length > 0) {
      const tmp = queue.shift();
      if (bfsVisited[tmp]) {
        continue;
      }
      bfsAnswer.push(tmp);
      bfsVisited[tmp] = true;

      for (let t = 0; t < graph[tmp][t]; t++) {
        if (!bfsVisited[graph[tmp][t]]) {
          queue.push(graph[tmp][t]);
        }
      }
    }

    return;
  }

  dfs(n, v, graph);
  bfs(n, v, graph);
  dfsAnswer = dfsAnswer.join(" ");
  console.log(dfsAnswer);
  bfsAnswer = bfsAnswer.join(" ");
  console.log(bfsAnswer);
}

const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
rl.on("line", function (line) {
  input.push(line);
}).on("close", function () {
  const [n, m, v] = input[0].split(" ").map(Number);
  let graph = new Array(n + 1);
  for (let i = 0; i < n + 1; i++) {
    graph[i] = new Array();
  }

  for (let i = 0; i < m; i++) {
    const [start, end] = input[i + 1].split(" ").map(Number);
    graph[start].push(end);
    graph[end].push(start);
  }

  graph.map((arr) => arr.sort((a, b) => a - b));

  solution(n, v, graph);

  process.exit();
});
